New commit makes tests unable to run

**Describe the bug**
A clear and concise description of what the bug is.

I installed `neotest-zig` today with commit 3a1c07319e826ad47b8f59824f6a60205a5e11fd. I tried to run tests but got the following output:

```
test
└─ run test
   └─ zig test Debug native 17 errors
/home/ziontee113/.local/share/nvim/lazy/neotest-zig/zig/neotest_runner.zig:24:14: error: root struct of file 'debug' has no member named 'lockStdErr'
    std.debug.lockStdErr();
    ~~~~~~~~~^~~~~~~~~~~
/usr/lib/zig/std/debug.zig:1:1: note: struct declared here
const std = @import("std.zig");
^~~~~
referenced by:
    log__anon_7581: /usr/lib/zig/std/log.zig:125:22
    err__anon_3288: /usr/lib/zig/std/log.zig:175:16
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
/home/ziontee113/.local/share/nvim/lazy/neotest-zig/zig/neotest_runner.zig:24:14: error: root struct of file 'debug' has no member named 'lockStdErr'
    std.debug.lockStdErr();
    ~~~~~~~~~^~~~~~~~~~~
/usr/lib/zig/std/debug.zig:1:1: note: struct declared here
const std = @import("std.zig");
^~~~~
/home/ziontee113/.local/share/nvim/lazy/neotest-zig/zig/neotest_runner.zig:24:14: error: root struct of file 'debug' has no member named 'lockStdErr'
    std.debug.lockStdErr();
    ~~~~~~~~~^~~~~~~~~~~
```

Any commit before that works fine for me.

## **Steps To Reproduce**
1. Go to dir
2. `zig init`
3. Go to `main.zig`, run `neotest.run.run(vim.fn.expand("%"))`

## **Screenshots**

![Screenshot from 2024-06-08 06-23-58](https://github.com/lawrence-laz/neotest-zig/assets/102876811/1fb1ccd9-891a-48ff-89a3-461a80d71403)

## **Environment**
 - OS: Arch Linux
 - Zig version: 0.12
 - Nvim version: 0.10
 - Neotest version: HEAD
 - Neotest Zig version: HEAD

--- a/zig/neotest_runner.zig
+++ b/zig/neotest_runner.zig
@@ -21,10 +21,28 @@ pub fn runnerLogFn(
 
     const prefix = "[" ++ comptime level.asText() ++ "] ";
 
-    std.debug.lockStdErr();
-    defer std.debug.unlockStdErr();
+    lockStderr();
+    defer unlockStdErr();
     const stderr = std.io.getStdErr().writer();
     nosuspend stderr.print(prefix ++ format ++ "\n", args) catch return;
+}
+
+fn lockStderr() void {
+    if (@hasDecl(std.debug, "lockStdErr")) {
+        std.debug.lockStdErr();
+    } else {
+        // v0.12.0 compatability
+        std.debug.getStderrMutex().lock();
+    }
+}
+
+fn unlockStdErr() void {
+    if (@hasDecl(std.debug, "unlockStdErr")) {
+        std.debug.unlockStdErr();
+    } else {
+        // v0.12.0 compatability
+        std.debug.getStderrMutex().unlock();
+    }
 }
 
 const STATUS_FAILED = "failed";

